Updated the code to work with current devel Anaconda.

matejak opened the pull request:
The Anaconda API between the `F28` and current `master` branch differs, so the OAA code also needs to be updated.

What has been done:
* The `ksdata.firewall` interface has been updated to use the `firewall_proxy` interface. There is a slight nuisance here that the proxy interface supports get/set methods, but not set operations s.a. add, difference, etc., so the code doesn't do a 100% right thing. I have decided to leave it like it is for the time being, as we are under time pressure, and the firewall settings are not used in our security content.
* The `ksdata.rootpw` has been replaced by `users_proxy` interface.
* The `storage.bootloader` has been replaced by the `bootloader_proxy` interface. Note: The interface seems to have expanded, so we can aim to set the bootloader password in the future.

What has not been done:
* The `storage.mountpoints` interface conversion (As confirmed by Anaconda people, this is OK, as `storage` in this context points to `blivet`, which has the same API as we are used to).
* The `ksdata.packages` interface conversion (As confirmed by Anaconda people, this is OK).
* The `ksdata.addons.com_redhat_kdump` interface that queries settings of the Kdump addon (As confirmed by Anaconda people, this is OK, we may have to check with Kdump people concerning their API, but it is extremely unlikely that they change it).

pep8speaks commented:
Hello @matejak! Thanks for submitting the PR.

 - There are no PEP8 issues in the file [`org_fedora_oscap/rule_handling.py`](https://github.com/matejak/oscap-anaconda-addon/blob/b302f393506415809381cfc13378329b013788a3/org_fedora_oscap/rule_handling.py) !

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/a3f86eda-6103-4de6-820c-053c861d53df) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/a3f86eda-6103-4de6-820c-053c861d53df) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/6adc6678-9096-4a5a-a164-3dfd5023ab4d) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/6f237026-8fec-4dc2-834d-e8856e5bc088) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/6f237026-8fec-4dc2-834d-e8856e5bc088) completed: **No new issues**

yuumasato commented on org_fedora_oscap/rule_handling.py:
```diff
@@ -895,25 +907,26 @@ def __str__(self):
         if self._firewall_default_enabled is None:
             # firewall default startup setting
-            self._firewall_default_enabled = ksdata.firewall.enabled
+            self._firewall_default_enabled = firewall_proxy.FirewallMode
```
I would suggest to call this variable `_firewall_detault_state`, the current name is confusing.

yuumasato replied:
Or `_firewall_default_mode`.

yuumasato commented on org_fedora_oscap/rule_handling.py:
```diff
@@ -938,48 +951,51 @@ def eval_rules(self, ksdata, storage, report_only=False):
             messages.append(RuleMessage(self.__class__,
                                         common.MESSAGE_TYPE_INFO, msg))
+        if not report_only:
+            firewall_proxy.SetEnabledServices(self._add_svcs)
```
I couldn't exactly find in documentation, but from [source code](https://github.com/rhinstaller/anaconda/tree/master/pyanaconda/modules/network/firewall), it looks like the new interface doesn't append, it just sets the new list of services.

So this should probably contain both the newly enabled and already enabled services.

jkonecny12 replied:
Yes the documentation is outdated for the current code, sorry for that. And again yes, as you said this should set a list not append.

yuumasato commented on org_fedora_oscap/rule_handling.py:
```diff
@@ -938,48 +951,51 @@ def eval_rules(self, ksdata, storage, report_only=False):
             messages.append(RuleMessage(self.__class__,
                                         common.MESSAGE_TYPE_INFO, msg))
+        if not report_only:
+            firewall_proxy.SetEnabledPorts(self._add_ports)
```
The same comment as in `SetEnabledServices`.

yuumasato commented on org_fedora_oscap/rule_handling.py:
```diff
@@ -938,48 +951,51 @@ def eval_rules(self, ksdata, storage, report_only=False):
             messages.append(RuleMessage(self.__class__,
                                         common.MESSAGE_TYPE_INFO, msg))
+        if not report_only:
+            firewall_proxy.SetTrusts(self._add_trusts)
```
The same comment as in `SetEnabledServices`.

yuumasato commented on org_fedora_oscap/rule_handling.py:
```diff
@@ -990,28 +1006,30 @@ def eval_rules(self, ksdata, storage, report_only=False):
             messages.append(RuleMessage(self.__class__,
                                         common.MESSAGE_TYPE_INFO, msg))
+        if not report_only:
+            firewall_proxy.SetDisabledServices(self._removed_svcs)
```
The same comment as in `SetEnabledServices`.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/f5217c4c-966d-4337-a3f1-066bef0467ce) was created.

matejak commented:
I have rebased, and incorporated fixes for flaws pointed by @yuumasato . The code needs a more serious refactoring, but that is out of scope for this PR.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/f5217c4c-966d-4337-a3f1-066bef0467ce) completed: **No new issues**

yuumasato commented on org_fedora_oscap/rule_handling.py:
```diff
@@ -937,49 +950,55 @@ def eval_rules(self, ksdata, storage, report_only=False):
             messages.append(RuleMessage(self.__class__,
                                         common.MESSAGE_TYPE_INFO, msg))
+        if not report_only:
+            all_services = list(self._add_svcs.union(set(firewall_proxy.EnabledServices)))
```
Would it be better to put together add `self._added_svcs` to `firewall_proxy.EnabledServices`? We know they will be disjoint sets.

matejak replied:
I would say that it's the same.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/a79e9e6a-5a0c-4828-a3b6-f502c76293f8) was created.

matejak commented:
@yuumasato I have incorporated changes to the revert functionality.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/a79e9e6a-5a0c-4828-a3b6-f502c76293f8) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/8aeb8e3b-fe6d-405e-b762-99e226fb2a4a) was created.

matejak commented on org_fedora_oscap/rule_handling.py:
```diff
@@ -990,52 +1017,56 @@ def eval_rules(self, ksdata, storage, report_only=False):
-            if trust in ksdata.firewall.trusts:
-                ksdata.firewall.trusts.remove(trust)
+        all_trusts = firewall_proxy.Trusts
+        orig_trusts = set(all_trusts).difference(self._new_ports_to_add)
```
good catch! Fixed.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/8aeb8e3b-fe6d-405e-b762-99e226fb2a4a) completed: **No new issues**

yuumasato commented:
@matejak Thanks for these updates!
